src/Infrastructure/Security/Authenticator/ProConnectAuthenticator.php

```diff
@@ -0,0 +1,143 @@
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authenticator\AbstractAuthenticator;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Se référer à la documentation technique
+ * https://github.com/numerique-gouv/proconnect-documentation/blob/main/doc_fs/implementation_technique.md
+ */
+class ProConnectAuthenticator extends AbstractAuthenticator
+{
+    public function __construct(
+        private HttpClientInterface $httpClient,
+        private UrlGeneratorInterface $urlGenerator,
+        private CommandBusInterface $commandBus,
+        private string $proConnectClientId,
+        private string $proConnectClientSecret,
+        private string $proConnectDomain,
+    ) {
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $request->attributes->get('_route') === 'pro_connect_callback';
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        try {
+            $session = $request->getSession();
+
+            // Vérification du state
+            $receivedState = $request->query->get('state');
+            $originalState = $session->get('oauth2_state');
+
+            if (empty($receivedState) || $receivedState !== $originalState) {
+                throw new AuthenticationException('Invalid state parameter');
+            }
+
+            // Vérification du code
+            $code = $request->query->get('code');
+            if (empty($code)) {
+                throw new AuthenticationException('No authorization code provided');
+            }
+
+            // Échange du code contre un token
+            $tokenData = $this->exchangeCodeForToken($code);
+            // Vérification de la réponse du token
+            if (!isset($tokenData['access_token']) || !isset($tokenData['id_token'])) {
+                throw new AuthenticationException('Invalid token response');
+            }
+
+            // Stockage de l'id_token pour la déconnexion
+            $session->set('id_token', $tokenData['id_token']);
+
+            // Récupération des infos utilisateur
+            $userInfo = $this->getUserInfo($tokenData['access_token']);
+
+            // Vérification des données utilisateur
+            if (!isset($userInfo['email'])) {
+                throw new AuthenticationException('Email not found in user info');
+            }
+
+            // Creation du user proConnect s'il n'existe pas
+            $this->commandBus->handle(new CreateProConnectUserCommand($userInfo['email'], $userInfo));
```

Comment:
Décomposer le userInfo ici ? Car on laisse fuiter dans la couche Application le nom des champs du userInfo...

```suggestion
            $this->commandBus->handle(new CreateProConnectUserCommand(
                $userInfo['email'],
                givenName: $userInfo['given_name'],
                usualName: $userInfo['usual_name'],
                siret: $userInfo['siret'],
            ));
```